### PR TITLE
Fix test for subject name change

### PR DESCRIPTION
**Pull request by joannanosek**
Summary of changes:

- add asserts of subject name in e2e test for endpoint creation
- move updating of subject name test to functional tests

### CODE REVIEW

File: tests/e2e_tests/inference_endpoints/test_inference_endpoints.py [hunk 1]
```diff
@@ -195,21 +196,50 @@ def test_prediction_with_certificates_and_wrong_subject_name():
...
-    assert running is True
+    updated = wait_ingress_setup(ING_NAMESPACE, TENANT_NAME, ENDPOINT_NAME, start_time)
+    assert updated is True
+    subject_name = get_ingress_subject_name(ENDPOINT_NAME, TENANT_NAME)
+    assert subject_name == 'CN=client'
     return endpoint_response
 
 
+def wait_ingress_setup(ingress_namespace, endpoint_namespace, endpoint_name, start_time):
+    ingress_pod_name = get_ingress_pod_name(ingress_namespace)
```
Comment (mzegla) [inline tests/e2e_tests/inference_endpoints/test_inference_endpoints.py]: There's no waiting between update call and calling this method. Isn't it possible that Kubernetes won't create updated ingress pod before `get_ingress_pod_name` call? I assume that if it won't, we will get Exception in 
`logs = get_logs_of_pod(ingress_namespace, ingress_pod_name, since_seconds=timer)` below.  Then in every iteration we get into except block, log the same error and wait 10 sec in every iteration. Is that right or am I missing something?

File: tests/conftest.py [hunk 2]
```diff
@@ -358,3 +360,24 @@ def resource_quota(api_instance, quota={}, namespace=TENANT_NAME):
     body = client.V1ResourceQuota(spec=resource_quota_spec, metadata=name_object)
     api_instance.create_namespaced_resource_quota(namespace=namespace, body=body)
     return quota
+
+
+def get_endpoint_ingress(name, namespace):
+    extensions_api_instance = client.ExtensionsV1beta1Api(config.load_kube_config())
+    api_response = extensions_api_instance.read_namespaced_ingress(name, namespace)
```
Comment (mzegla) [inline tests/conftest.py]: Any handling in case of error?

File: tests/conftest.py [hunk 3]
```diff
@@ -358,3 +360,24 @@ def resource_quota(api_instance, quota={}, namespace=TENANT_NAME):
...
+    return api_response
+
+
+def get_ingress_subject_name(name, namespace):
+    metadata = get_endpoint_ingress(name, namespace).metadata
```
Comment (mzegla) [inline tests/conftest.py]: Any handling in case of error?

File: tests/e2e_tests/inference_endpoints/test_inference_endpoints.py (hunk 1 above)
Comment (joannanosek) [reply tests/e2e_tests/inference_endpoints/test_inference_endpoints.py]: Kubernetes is not creating new pod for ingress, so there's no need to handle this situation

File: tests/conftest.py (hunk 2 above)
Comment (joannanosek) [reply tests/conftest.py]: Hmm this is confusing, as in the beginning we agreed to an assumption that in tests function everything works fine so we are not going to catch errors (even [here](https://github.com/IntelAI/inference-model-manager/blob/master/tests/conftest.py#L72)) and after that a few calls appeared which has an error handling. So it is a matter of what path we want to go

File: tests/conftest.py (hunk 3 above)
Comment (joannanosek) [reply tests/conftest.py]: Same as above

File: tests/e2e_tests/inference_endpoints/test_inference_endpoints.py [hunk 4]
```diff
@@ -195,21 +196,50 @@ def test_prediction_with_certificates_and_wrong_subject_name():
...
+    return updated
+
+
+def retrieve_log_line_for_updated_ingress(logs, namespace, name):
+    logs = logs.splitlines()
+    for line in logs:
+        if namespace in line and name in line and "reason: 'UPDATE'" in line:
+            break
+        logs.pop(0)
+    for line in logs:
```
Comment (beataszelag) [inline tests/e2e_tests/inference_endpoints/test_inference_endpoints.py]: did you consider shorter version:
```
return 'Backend...' in logs
```

**Pull request merged**